src/lib/features/project/project-owners-read-model.type.ts

```diff
@@ -0,0 +1,29 @@
+import type { IProjectWithCount } from '../../types';
+
+export type SystemOwner = { ownerType: 'system' };
+export type UserProjectOwner = {
+    ownerType: 'user';
+    name: string;
+    email?: string;
+    imageUrl?: string;
+};
+export type GroupProjectOwner = {
+    ownerType: 'group';
+    name: string;
+};
+type ProjectOwners =
+    | [SystemOwner]
+    | Array<UserProjectOwner | GroupProjectOwner>;
+
+export type ProjectOwnersDictionary = Record<string, ProjectOwners>;
+
+export type IProjectWithCountAndOwners = IProjectWithCount & {
+    owners: ProjectOwners;
+};
+
+export interface IProjectOwnersReadModel {
+    getAllProjectOwners(): Promise<ProjectOwnersDictionary>;
```

Comment:
Why is this part of the interface? I don't think this should ever be called by an external module?